services/catalog/src/simcore_service_catalog/api/routes/services_resources.py

```diff
@@ -25,104 +33,175 @@
-                f"{service_key}:{service_version}",
-            )
-            continue
-
-        if nano_cpu_limit := entry.value.get("Limits", {}).get("NanoCPUs"):
-            service_resources["CPU"].limit = nano_cpu_limit / 1.0e09
-        if nano_cpu_reservation := entry.value.get("Reservations", {}).get("NanoCPUs"):
-            service_resources["CPU"].reservation = nano_cpu_reservation / 1.0e09
-        if ram_limit := entry.value.get("Limits", {}).get("MemoryBytes"):
-            service_resources["RAM"].limit = ram_limit
-        if ram_reservation := entry.value.get("Reservations", {}).get("MemoryBytes"):
-            service_resources["RAM"].reservation = ram_reservation
-
-        _parse_generic_resource(
-            entry.value.get("Reservations", {}).get("GenericResources", []),
-            service_resources,
-        )
-
-    return service_resources
+SIMCORE_SERVICE_SETTINGS_LABELS: Final[str] = "simcore.service.settings"
+SIMCORE_SERVICE_COMPOSE_SPEC_LABEL: Final[str] = "simcore.service.compose-spec"
 
 
 @router.get(
     "/{service_key:path}/{service_version}/resources",
-    response_model=dict[ResourceName, ResourceValue],
+    response_model=ServiceResourcesDict,
     **RESPONSE_MODEL_POLICY,
 )
 @cached(
     ttl=DIRECTOR_CACHING_TTL,
     key_builder=lambda f, *args, **kwargs: f"{f.__name__}_{kwargs['service_key']}_{kwargs['service_version']}",
 )
 async def get_service_resources(
-    service_key: ServiceKey,
-    service_version: ServiceVersion,
+    service_key: DockerImageKey,
+    service_version: DockerImageVersion,
     director_client: DirectorApi = Depends(get_director_api),
-    default_service_resources: ServiceResources = Depends(
-        get_default_service_resources
-    ),
-):
+    default_service_resources: ResourcesDict = Depends(get_default_service_resources),
+) -> ServiceResourcesDict:
+    # TODO: --> PC: I'll need to go through that with you for function services,
     # cause these entries are not in ServiceDockerData
+
+    image_version = f"{service_key}:{service_version}"
     if is_function_service(service_key):
-        # NOTE: this is due to the fact that FastAPI does not appear to like pydantic models with just a root
-        return default_service_resources.dict()["__root__"]
-
-    service_labels: dict[str, Any] = cast(
-        dict[str, Any],
-        await director_client.get(
-            f"/services/{urllib.parse.quote_plus(service_key)}/{service_version}/labels"
-        ),
+        return ServiceResourcesDictHelpers.create_from_single_service(
+            image_version, default_service_resources
+        )
+
+    def _from_service_settings(
+        settings: List[SimcoreServiceSettingLabelEntry],
+    ) -> ResourcesDict:
+        # filter resource entries
+        resource_entries = filter(
+            lambda entry: entry.name.lower() == "resources", settings
+        )
+        # get the service resources
+        service_resources = deepcopy(default_service_resources)
```

Comment:
Q: why use deepcopy instead of copy(deep=True) ? like it was? what is the difference?

Reply:
Because previously I had a `pydantic` `BaseModel`. Now I have a `dict[...]` which contains pedantic derived models and other types. The dict's copy is shallow(and has no extra parameter for deep copies), I require a deep copy.